src/Components/Todo/TodoListItem.tsx

```diff
@@ -0,0 +1,166 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import { useContext } from 'react';
+import { Todo } from '../../types/Todo';
+import { TodoContext } from '../TodoContext/TodoContext';
+import classNames from 'classnames';
+import { useState } from 'react';
+import { deleteTodo, updateTodo } from '../../api/todos';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoListItem: React.FC<Props> = ({ todo }) => {
+  const { todosList, setTodosList, setErrorMessage } = useContext(TodoContext);
+
+  const { id, title, completed } = todo;
+
+  const [editable, setEditable] = useState(false);
+
+  const [newTitle, setNewTitle] = useState(title);
+
+  const [trimmedTitle, setTrimmedTitle] = useState(newTitle.trim());
+
+  const removeTodo = () => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    deleteTodo(id)
+      .then(() => {
+        setTodosList(todosList.filter(item => item.id !== id));
+      })
+      .catch(() => setErrorMessage('Unable to delete a todo'));
+  };
+
+  const setTodoStatus = () => {
+    const targetTodo = todosList.find(item => item.id === todo.id);
+
+    if (targetTodo) {
+      targetTodo.completed = !todo.completed;
+
+      updateTodo({ ...todo })
+        .then(() => {
+          setTodosList(
+            todosList.map(task =>
+              task.id === id
+                ? { ...todo, completed: targetTodo.completed }
+                : task,
+            ),
+          );
+        })
+        .catch(() => {
+          setErrorMessage('Unable to update a todo');
+        });
+    }
+  };
+
+  const changeStatus = () => {
+    setTodoStatus();
+  };
```

Comment:
redundant function wrapper - just use `setTodoStatus` instead of `changeStatus`
but you should pass value there, don't you?